packages/core/src/external/carequality/ihe-gateway-v2/monitor/store.ts

```diff
@@ -143,3 +281,22 @@ export function buildIheResponseKey({
   const requestChunkIdPart = requestChunkId ? `_${requestChunkId}` : "";
   return `${cxId}/${patientId}/${type}/${requestId}_${date}/${oid}${requestChunkIdPart}.xml`;
 }
+
+export function buildIheRequestKey({
+  type,
+  oid,
+  requestId,
+  day,
+  timestamp,
+  extension,
+}: {
+  day: string;
+  oid: string;
+  type: "xcpd" | "dq" | "dr";
+  requestId: string;
+  timestamp: string;
+  extension: "xml" | "json";
+}) {
+  const date = dayjs(timestamp).toISOString();
+  return `${day}/${oid}/${type}/${requestId}_${date}.${extension}`;
```

Comment:
we can change this if its not the easiest for debugging. My thinking here was break it down by day, then who sent it, then what kind of request, then if we have the requestId or timestamp we can use that to find it

Reply:
this is pretty good for now